### ISSUE TITLE
Trampoline version message UX problems

### ISSUE THREAD

**Issue by indirect** [bundler/bundler#4753]
Here's what I did:

```
$ echo 'source "https://rubygems.org"' > Gemfile

$ echo 'GEM
  remote: https://rubygems.org/
  specs:

PLATFORMS
  ruby

DEPENDENCIES

BUNDLED WITH
   1.12.0' > Gemfile.lock

$ dbundle install
Fetching: bundler-1.12.0.gem (100%)
You're running Bundler 1.13.0.rc.1 but this project uses 1.12.0. To update, run `bundle update --bundler`.
The Gemfile specifies no dependencies
Bundle complete! 0 Gemfile dependencies, 1 gem now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.

$ dbundle -v
1.12.0
```

After running `bundle install`, no output appeared for approximately 10 seconds. That's too long, and it makes it seem like Bundler has frozen. When you run `bundle install`, it should _immediately_ print a message like "Switching to locked Bundler version 1.12.0...". Then, if 1.12.0 still needs to be installed, we can optionally print something like "Downloading and installing Bundler 1.12.0...".

The later "You're running Bundler.." message should _not_ appear. We already promised (the first time they saw that message) that we wouldn't show that message again!

Finally, when you run `bundle -v` or `bundle version`, we need to print out the version of Bundler that is actually installed, not the version of Bundler that is in the lockfile. The output should be `1.13.0.rc.1`. If they need to know which version of Bundler this bundle is locked to, they can run `bundle show bundler`, and it will print a path that includes the version.

/cc @RochesterinNYC @b-ggs @segiddins

**Comment by segiddins** [comment]
Why shouldn't bundle -v print out the version that will run?

**Comment by segiddins** [comment]
@indirect I believe it prints twice since the trampoline is unable to access Bundler.settings -- it won't print twice when trampolining to a version >= 1.13.0.rc.1

**Comment by segiddins** [comment]
I also personally think it shouldn't print anything different when installing, as that's simply just noise and is liable to break anything that depends upon bundler output

**Comment by indirect** [comment]
@segiddins It's a non-starter to run `gem install bundler`, see "Installed Bundler 1.13" and then have `bundle -v` print an older version. If you run `bundle -v`, it should print the version of the gem that you have installed and that was invoked by the `bundle` command. If you want to also show the version that will be used to run other commands, we could print that as well, maybe something like `1.13.0 (Gemfile locked to 1.12.5)`.

I guess I wasn't very clear about this before, and I'm sorry about that. There are two different kinds of trampoline messages that we should print:

1. We should print an info-level message with wording something like "Switching to locked Bundler 1.12.5..." every single time we trampoline.
1. The first time you run a `bundle` command inside a directory locked to an older version of Bundler than the version loaded by `bundle`, we should print a warning-level message along these lines:
  > You're running Bundler 1.13.0.rc.1 but this project uses 1.12.0. To update, run `bundle update --bundler`. You won't see this message again until the next time you upgrade Bundler.
  (I think this part is already implemented, except that it prints said warning even if the locked version and the running version are the same version. That issue is #4752.)

In addition to those two things, it's not acceptable for any `bundle` command to hang for 12 seconds without a single character of output. We need to say something like "Switching to locked Bundler 1.12.5..." or "Installing locked Bundler 1.12.5..." immediately, so that the user knows that their command has not hung.

Ultimately, this is about making it clear to a person who typed `gem install bundler && bundle install` what in the actual fuck is happening when the result is Bundler 1.12 code running, and not Bundler 1.13 code running. This change is completely unprecedented and undocumented (and let's be real, most people won't read the announcement or the docs). We need to make it clear to someone who did absolutely nothing except run `gem install bundler` what is going on when they run `bundle install`.

**Comment by b-ggs** [comment]
I agree with printing out something before it starts fetching an older bundler gem. Recently, I was working at a place that had an intermittent connection, and I was wondering why `dbundle` was taking so long to run. Turns out, it was attempting to get an older version of `bundler`, and was failing.

It would have been nice to know that before I `Ctrl-c`'d out.

---

> The later "You're running Bundler.." message should not appear. We already promised (the first time they saw that message) that we wouldn't show that message again!

As for this, it's because, as @segiddins mentioned, we can't access `Bundler.settings` from the trampoline. That message will always be printed only if the bundle is locked to a version `< 1.13.pre`, but if the bundle is locked to something after that version, it will only be printed once.

See this: https://github.com/bundler/bundler/blob/master/lib/bundler/postit_trampoline.rb#L52

**Comment by indirect** [comment]
@b-ggs cool, let's add a message like "Installing locked Bundler version 1.12.5..." before the install starts, so that users can tell what's going on.

For the "You're running Bundler..." message, we can wait to print it until much later when we do have access to `Bundler.settings`. The only message that needs to print out at `postit_trampoline.rb:52` is something like `puts "Switching to locked Bundler version 1.12.5..."`.

**Comment by indirect** [comment]
The second half of this issue has been resolved by #4740, so all that's left is to print `Installing locked Bundler version 1.12.5...` before hanging with no output. Getting very close!

**Comment by segiddins** [comment]
```diff
diff --git a/lib/bundler/postit_trampoline.rb b/lib/bundler/postit_trampoline.rb
index 010293e..5a68bb2 100644
--- a/lib/bundler/postit_trampoline.rb
+++ b/lib/bundler/postit_trampoline.rb
@@ -18,6 +18,7 @@ installed_version &&= Gem::Version.new(installed_version)
 
 if !version.satisfied_by?(installed_version)
   begin
+    puts "Installing locked Bundler version #{version}"
     installer = BundlerVendoredPostIt::PostIt::Installer.new(version)
     installer.install!
   rescue => e
```
